### ISSUE TITLE
Deploy to AWS Elastic Beanstalk

### ISSUE THREAD

**Issue by mkloubert** [mkloubert/vs-deploy#32]
[Mauro Di Iasio](https://marketplace.visualstudio.com/items?itemName=mkloubert.vs-deploy#review-details):

> I only have a problem:
> "targets": [
> {
> "type": "zip",
> "name": "Sample.zip",
> "description": "Example ZIP per AWS",
> "target": "D:/desktop/Deploy"
> }
> ]
> 
> When I deploy zip file on Elastic Beanstalk, it seems fine but don't do the upload!
> 
> Have I to use a different target ?

I am not sure: The [zip target](https://github.com/mkloubert/vs-deploy/wiki/target_zip) only creates a ZIP files in a "local" directory.

Can you (or someone else) explain how you upload/deploy a ZIP file (content) to [AWS Elastic Beanstalk](https://aws.amazon.com/de/elasticbeanstalk/)?

**Comment by FerrarisIV** [comment]
Thank you to reply me dear Marcel, I will try to explain you

To upload my Web app in AWS Amazon, in Visual Studio I simply do this: build -> Publish Web app (publish method: Web Deploy Package)
VS will create a zip folder in which is the complete package

Here is the guide: http://docs.aws.amazon.com/elasticbeanstalk/latest/dg/applications-sourcebundle.html

**Creating a Source Bundle for a .NET Application**
....
If you need to manually create a source bundle for your .NET application, you cannot simply create a Zip file that contains the project directory. You must create a web deployment package for your project that is suitable for deployment to Elastic Beanstalk. There are several methods you can use to create a deployment package:


How can I do this by your extension in Visual Code ?

Thanks in advance
Mauro

**Comment by mkloubert** [comment]
@FerrarisIV 

**I have released version 5.15.0 that is now albe to define a custom output filename for a ZIP file.**

That should make it possible to create a ZIP file with the same name and use a [webdeploy](https://github.com/mkloubert/vs-deploy/wiki/targetoperations#webdeploy-) post operation:

```json
{
    "deploy": {
        // ...
        
        "targets": [
            {
                "type": "zip",
                "name": "Sample.zip",
                "description": "Example ZIP per AWS",
                "target": "D:/desktop/Deploy"
                "fileName": "my_aws_app.zip",

                "deployed": [
                    {
                        "type": "webdeploy",
                        "exec": "C:/Program Files/IIS/Microsoft Web Deploy V3/msdeploy.exe",

                        "verb": "sync",
                        "source": "lm/w3svc/1",
                        "package": "D:/desktop/Deploy/my_aws_app.zip"
                    }
                ]
            }
        ]
        
        // ...
    }
}
```

This is only an example for the 1st example from [here](https://www.iis.net/learn/publish/using-web-deploy/packaging-and-restoring-a-web-site).

Please check out the required parameters and the path where `msdeploy.exe` is installed on your system.

**Comment by FerrarisIV** [comment]
Thanks again, I will try it next week!!
